File: managment/src/main/java/com/prestabanco/managment/services/ClientService.java
package com.prestabanco.managment.services;


import com.prestabanco.managment.entities.ClientEntity;
import com.prestabanco.managment.repositories.ClientRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service
public class ClientService {
    @Autowired
    ClientRepository clientRepository;

    // Get all clients in ClientRepository
    public ArrayList<ClientEntity> getClients() {
        return (ArrayList<ClientEntity>) clientRepository.findAll();
    }

    // Save a client
    public ClientEntity saveClient(ClientEntity client) {
        return clientRepository.save(client);
    }

    // Get a client by id
    public ClientEntity getClientById(Long id) {
        return clientRepository.findById(id).get();
    }

    // get a client by Rut
    public ClientEntity getClientByRut(String rut) {
        return clientRepository.findByRut(rut);
    }
    public ClientEntity updateClient(ClientEntity client) {
        return clientRepository.save(client);
    }

    // delete a client by id
    public boolean deleteClient(Long id) throws Exception {
        try{
            clientRepository.deleteById(id);
            return true;
        } catch (Exception e) {
            throw new Exception(e.getMessage());
        }
    }

}

File: front/interface/src/services/client.service.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/v1/clients/";

// obtain all clients
const getAllClients = () => {
    return axios.get(API_URL);
};

// obtain a client by id
const getClientById = (id) => {
    return axios.get(API_URL + id);
};

// register a new client
const createClient = (client) => {
    return axios.post(API_URL, client);
};

// update a client
const updateClient = (client) => {
    return axios.put(API_URL, client);
};

// delete a client by id
const deleteClient = (id) => {
    return axios.delete(API_URL + id);
};

export default {
    getAllClients,
    getClientById,
    createClient,
    updateClient,
    deleteClient,
};
